modules/utask/encoding.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utask
+
+import (
+	"bytes"
+	"encoding/gob"
+
+	"go.uber.org/fx/ulog"
+)
+
+// Encoding is capable of encoding and decoding objects
+type Encoding interface {
+	Register(interface{}) error
+	Marshal(interface{}) ([]byte, error)
+	Unmarshal([]byte, interface{}) error
+}
+
+// NopEncoding is a noop encoder
+type NopEncoding struct {
+}
+
+// Register implements the Encoding interface
+func (g *NopEncoding) Register(obj interface{}) error {
+	return nil
+}
+
+// Marshal implements the Encoding interface
+func (g *NopEncoding) Marshal(obj interface{}) ([]byte, error) {
+	return []byte{}, nil
+}
+
+// Unmarshal implements the Encoding interface
+func (g *NopEncoding) Unmarshal(data []byte, obj interface{}) error {
+	return nil
+}
+
+// GobEncoding encapsulates gob encoding and decoding
+type GobEncoding struct {
+}
+
+// Register implements the Encoding interface
+func (g *GobEncoding) Register(obj interface{}) error {
+	gob.Register(obj)
+	return nil
+}
+
+// Marshal encodes an object into bytes
+func (g *GobEncoding) Marshal(obj interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	if err := enc.Encode(obj); err != nil {
```

Comment:
This probably isn't the right place to log this -- the caller should be expecting that the task library does error handling but not logging, otherwise it would pass it a logger. In other words, just do:

```go
err := enc.Encode(obj)
return buf.Bytes(), err
```